Application/hooks/useFriendList.tsx

```diff
@@ -54,10 +55,12 @@ export function useFriendList({ lastFetched }: FetchStatusProps) {
 
           if (response.ok) {
             const data = await response.json();
-            setIDs(data.friendList); // friendList is now an array if user IDs
-          } else {
+            setIDs(data.friendList); // If IDs are fetched
+            if(data.friendList.length === 0) setLoading(false); // If no IDs, then we're not loading anymore
+        } else {
             console.error('Failed to fetch friend list');
-          }
+            setLoading(false); // If the fetch fails, we're not loading anymore
+        }
         } catch (error) {
```

Comment:
- While error logging is implemented for failed fetch requests, there is no mechanism to communicate these errors to the user. Consider adding error handling logic to provide feedback to users when fetch requests fail, such as displaying an error message or retry option.